server/controllers/articleController.js

```diff
@@ -91,15 +104,27 @@ export default class ArticleController {
   static updateArticle(req, res) {
     const imgUrl = (req.file ? req.file.secure_url : '');
     req.body.imgUrl = imgUrl;
-    Article.update(req.body, {
-      where: { slug: req.params.slug, userId: req.user.id },
-      returning: true,
-    }).then((article) => {
-      if (article[0] === 0) {
-        res.status(404).json({ message: 'article does not exist', success: 'false' });
-      } else {
-        res.status(200).json({ message: 'article updated successfully', success: 'true', article });
-      }
+    getTagIds(req, res).then((tagIds) => {
+      Article.findOne({
+        where: { slug: req.params.slug, userId: req.user.id },
+      }).then((article) => {
+        if (article === null) {
+          res.status(404).json({ message: 'article does not exist', success: false });
+        } else {
+          article.update(req.body)
+            .then((updatedArticle) => {
```

Comment:
This change here gets the ids of the tags sent along with the request. Then the article is found, updated and tagged. Then, the article is returned with the tags.